### ISSUE TITLE
Referencing count created resources from another module in map array

### ISSUE THREAD

**Issue by hybmg57** [hashicorp/terraform#25243]
### Terraform Version
0.12.26

### Terraform Configuration Files
Modules in path `./ecs/ecs-service/*.tf`
```hcl
resource "aws_lb_target_group" "app" {
  count = length(var.target_groups)
  name = "test-${var.env}"
  port = var.target_groups[count.index].app_port
  protocol = "HTTP"
  vpc_id = var.vpc_id
  ...
}

output "app_target_group_arns" {
  value = aws_lb_target_group.app.*.arn
}
```

parent module `./ecs`
```hcl
module "ecs-service" {
  source = "./ecs-service"
  ...
}

module "alb" {
  source = "./ecs/alb"
  rules = [
    {
      priority : 100,
      target_group_arn : element(module.ecs-service.app_target_group_arns, 0),
      condition_field : "host-header",
      condition_values : [element(module.alb.alb_record_fqdn, 0)]
    },
    {
      priority : 101,
      target_group_arn : element(module.ecs-service.app_target_group_arns, 1),
      condition_field : "host-header",
      condition_values : [element(module.alb.alb_record_fqdn, 1)]
    }
  ]
}
```

### Debug Output
```
Call to function "element" failed: cannot use element function with an empty list
```

### Expected Behavior
module ecs-service targetgroup should wait until alb rules are deleted first so alb module doesn't reference elements that does not exist.

### Actual Behavior
errors with `Call to function "element" failed: cannot use element function with an empty list`

### Steps to Reproduce
run `terraform destroy` with above config

### References
https://github.com/hashicorp/terraform/issues/11210

**Comment by danieldreier** [comment]
@apparentlymart I think that this is basically an instance of https://github.com/hashicorp/terraform/issues/25162 ("non-obvious ordering of functions and resources") - does it look that way to you?

**Comment by jbardin** [comment]
This looks like the result of the variable values being evaluated during destroy, which should be one of the situations fixed in 0.13. 

@hybmg57 There's not enough information here for me to reproduce the issue, but if you have the ability to test with the latest 0.13 beta, we can see if my hunch is correct and this is fixed already.

**Comment by mildwonkey** [comment]
I am going to close this issue due to inactivity.

If there is still a question, I recommend the [the community forum](https://discuss.hashicorp.com/c/terraform-core), where there are far more people available to help. If there is a bug or you would like to make a feature request, please open a new issue and fill out the template.
Thanks!
